### ISSUE TITLE
Memory leak with no stored references

### ISSUE THREAD

**Issue by appellation** [nodejs/help#2541]
* **Node.js Version**: 13.10.1
* **OS**: Windows 10 (1903)
* **Scope (install, code, runtime, meta, other?)**: runtime
* **Module (and version) (if relevant)**: util?

I own the [lavalibs/encoding](https://github.com/lavalibs/encoding) repository and I am encountering a strange memory leak that I cannot resolve. This package consists of two pure functions that de/serialize a specialized encoding, so it does not allocate any state outside of structures needed to perform the de/serialization. One of my users was investigating this problem and wrote a script (below) that I was able to use to verify the memory leak. This script consistently resulted in memory usage over 400MB.

I eliminated the `base64-js` package as a source for the memory leak by running only the `Base64.toByteArray` function. Likewise I also verified that using `Buffer.alloc` results in expected memory usage (multiple allocs get GC'd as expected). I also ran the script with `--expose-gc` and explicitly triggered GC before generating the heapdump. The heapdump shows a lot of functions and arrays in memory, but I wasn't able to determine any useful information from that.

I am confused about how this script is causing a memory leak, since all of the decoded data should be marked for GC immediately after the loop iteration. I'm hypothesizing that the memory leak is related to either the `TextDecoder` (used to read binary data into the decoded structure) or `DataView` (used to read specific slices from the input `Uint8Array`) potentially keeping some sort of circular reference preventing GC, but I was unable to determine anything more specific than that.

I know that investigating memory leaks and heapdumps is not a fun or exciting experience, but I'm really lost on what could be causing this issue. Thank you for any insight you can provide.

```js
const { getHeapSnapshot } = require('v8');
const { createWriteStream } = require('fs');

const track = "QAAAkwIANFNrcmlsbGV4ICYgSGFic3RyYWt0IC0gQ2hpY2tlbiBTb3VwIFtPZmZpY2lhbCBBdWRpb10ABU9XU0xBAAAAAAADLIAACzIyTVdyV1BWX1FNAAEAK2h0dHBzOi8vd3d3LnlvdXR1YmUuY29tL3dhdGNoP3Y9MjJNV3JXUFZfUU0AB3lvdXR1YmUAAAAAAAAAAA==";

function reportMemory() {
  console.log(process.memoryUsage().heapUsed / 1024 / 1024, 'MB');
}

const { decode } = require('@lavalink/encoding');

for (let i = 0; i <= 100000; i++) {
  decode(track);
}

console.log('done');

setTimeout(() => {
  gc(); // only with --expose-gc flag
  reportMemory();
  getHeapSnapshot().pipe(createWriteStream('./heap.heapsnapshot'));
}, 1000);
```

*Edit:* note where we explicitly triggered GC

**Comment by Hakerh400** [comment]
Simplified code without any external dependency:

```js
'use strict';

const fs = require('fs');
const util = require('util');
const v8 = require('v8');

for(let i = 0; i !== 5e5; i++){
  new util.TextDecoder().decode(Buffer.from([
     83, 107, 114, 105, 108, 108, 101, 120,  32,  38,
     32,  72,  97,  98, 115, 116, 114,  97, 107, 116,
     32,  45,  32,  67, 104, 105,  99, 107, 101, 110,
     32,  83, 111, 117, 112,  32,  91,  79, 102, 102,
    105,  99, 105,  97, 108,  32,  65, 117, 100, 105,
    111,  93
  ]));
}

setTimeout(() => {
  gc();
  console.log(process.memoryUsage().heapUsed / (1 << 20), 'MB');
  v8.getHeapSnapshot().pipe(createWriteStream('./heap.heapsnapshot'));
}, 1e3);
```

**Comment by Hakerh400** [comment]
This eventually crashes the process:

```js
while(1) new util.TextDecoder().decode(Buffer.from('abcde'));
```

Seems like a memory leak in `util.TextDecoder`

**Comment by appellation** [comment]
@Hakerh400 Thank you for taking a look! I honestly hadn't considered investigating Node's internals for a memory leak. It doesn't look as if anyone has reported this issue or made a fix for it, and it still occurs on Node v13.11.0. I'll open this issue on the main NodeJS repository with this additional information.
